proof/crefine/Move_C.thy

```diff
@@ -1347,4 +1347,26 @@ lemma tcbSchedEnqueue_queued_queues_inv:
   apply normalise_obj_at'
   done
 
+lemma less_ucast_ucast_less':
+  "x < UCAST('b \<rightarrow> 'a) y \<Longrightarrow> UCAST('a \<rightarrow> 'b) x < y"
+  for x :: "'a::len word" and y :: "'b::len word"
+  by (clarsimp simp: order.strict_iff_order dual_order.antisym le_ucast_ucast_le)
+
+lemma ucast_up_less_bounded_implies_less_ucast_down':
+  assumes len: "LENGTH('a::len) < LENGTH('b::len)"
+  assumes bound: "y < 2 ^ LENGTH('a)"
+  assumes less: "UCAST('a \<rightarrow> 'b) x < y"
+  shows "x < UCAST('b \<rightarrow> 'a) y"
+  apply (rule le_less_trans[OF _ ucast_mono[OF less bound]])
+  using len by (simp add: is_down ucast_down_ucast_id)
+
+lemma ucast_up_less_bounded_iff_less_ucast_down':
```

Comment:
Should these be kept here, or moved to `lib/Word_Lib/Word_Lemmas_Internal.thy` as Gerwin suggested?

Reply:
I moved these to `Word_Lemmas_Internal` while rebasing. Fresh testboard below.